rafs/src/metadata/direct_v6.rs

```diff
@@ -179,9 +180,29 @@ impl DirectSuperBlockV6 {
             mapping: self.clone(),
             offset,
             blocks_count,
+            parent_inode: Cell::new(None),
+            name: Cell::new(None),
         }))
     }
 
+    // For RafsV6, we can't get the parent info of a non-dir file with its on-disk inode,
+    // so we need to pass corresponding parent info when constructing the child inode.
+    fn inode_wrapper_with_info(
+        &self,
+        nid: u64,
+        parent_inode: Inode,
+        name: OsString,
+    ) -> Result<Arc<OndiskInodeWrapper>> {
+        self.inode_wrapper(nid).map(|inode| {
+            let mut inode = inode;
+            // # Safety
+            // inode always valid
+            Arc::get_mut(&mut inode).unwrap().parent_inode = Cell::new(Some(parent_inode));
+            Arc::get_mut(&mut inode).unwrap().name = Cell::new(Some(name));
```

Comment:
Since `Cell` is not thread safe,  I suppose that it's not okay to assign `Cell` to a Arc which is a thread safe.